frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts

```diff
@@ -77,17 +84,33 @@ export const getEventDimensionsData = (
     });
   }
 
-  return dimensions;
+  return {
+    dimensions,
+    dimensionValue,
+  };
 };
 
 export const getEventColumnsData = (
   chartModel: CartesianChartModel,
   seriesIndex: number,
   dataIndex: number,
+  display: CardDisplayType,
 ) => {
-  const datum = chartModel.dataset[dataIndex];
   const seriesModel = chartModel.seriesModels[seriesIndex];
 
+  const isWaterfallTotal =
```

Comment:
Ideally we should avoid having display type specific code here. If we use cartesianChartModel as a base, we could extent its "raw" dataset with `Total` so that there is a correct datum with right data and it would work right away